tests/cpp/test_cpu_id_hash_map.cc

```diff
@@ -0,0 +1,60 @@
+#include <set>
+#include <gtest/gtest.h>
+#include <dgl/array.h>
+
+#include "../../src/array/cpu/cpu_id_hash_map.h"
+#include "./common.h"
+
+using namespace dgl;
+using namespace dgl::runtime;
+using namespace dgl::aten;
+
+namespace {
+
+template <typename IdType>
+void ConstructRandomSet(size_t size, size_t range,
+  std::vector<IdType>& id_vec) {
+    id_vec.resize(size);
+    std::srand(42);
+    for (size_t i = 0; i < size; i++) {
+        id_vec[i] = static_cast<IdType>(std::rand() % range);
+    }
+}
+
+template <typename IdType, size_t size, size_t range>
+void _TestIdMap() {
+    std::vector<IdType> id_vec;
+    ConstructRandomSet(size, range, id_vec);
+    std::set<IdType> id_set(id_vec.begin(), id_vec.end());
+    IdArray ids = VecToIdArray(id_vec, sizeof(IdType) * 8, CTX);
+    IdArray unique_ids = NewIdArray(size, CTX, sizeof(IdType) * 8);
+    CpuIdHashMap<IdType> id_map(CTX);
```

Comment:
No, It is only used in CPU. But there is another member in `DGLContext` called `device_id`, I'm not sure if it can always set to 0. If so, the argument can be removed.

Reply:
I think `device_id` is used for cuda device choose only

Reply:
I see, let me remove this parameter